source/lac/petsc_solver.cc

```diff
@@ -398,37 +389,10 @@ namespace PETScWrappers
     PetscErrorCode ierr = KSPSetType(ksp, KSPGMRES);
     AssertThrow(ierr == 0, ExcPETScError(ierr));
 
-    // set the restart parameter from the
-    // data. we would like to use the simple
-    // code that is commented out, but this
-    // leads to nasty warning and error
-    // messages due to some stupidity on
-    // PETSc's side: KSPGMRESSetRestart is
-    // implemented as a macro in which return
-    // statements are hidden. This may work
-    // if people strictly follow the PETSc
-    // coding style of always having
-    // functions return an integer error
-    // code, but the present function isn't
-    // like this.
-    /*
-        ierr = KSPGMRESSetRestart (ksp, additional_data.restart_parameter);
-        AssertThrow (ierr == 0, ExcPETScError(ierr));
-    */
-    // so rather expand their macros by hand,
-    // and do some equally nasty stuff that at
-    // least doesn't yield warnings...
-    int (*fun_ptr)(KSP, int);
-    ierr = PetscObjectQueryFunction(reinterpret_cast<PetscObject>(ksp),
```

Comment:
Yay!

Reply:
I don't know what that used to be but its definitely a normal function in `petscksp.h` these days.